### ISSUE TITLE
Report AND / OR expressions that do the same on left and right side

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#1000]
We should report AND / OR conditions that do the same comparison on both sides (left and right).

Example for AND:
```c#
if (x == 42 && x == 42)
```

Example for OR:
```c#
if (x == 42 || x == 42)
```

Both should be fixed to:
```c#
if (x == 42)
```

Be also aware of unneeded parenthesis on one side:
```c#
if (x == 42 || (x == 42))
```

**Comment by RalfKoban** [comment]
PR contained wrong ID, so this is not resolved
